ebpf/c/tracer-ebpf.c

```diff
@@ -802,6 +828,72 @@ int kprobe__tcp_retransmit_skb(struct pt_regs* ctx) {
     return handle_retransmit(sk, status);
 }
 
+SEC("kretprobe/inet_csk_accept")
+int kretprobe__inet_csk_accept(struct pt_regs* ctx) {
+    struct sock* newsk = (struct sock*)PT_REGS_RC(ctx);
+
+    if (newsk == NULL) {
+        return 0;
+    }
+
+    u64 zero = 0;
+    tracer_status_t* status = bpf_map_lookup_elem(&tracer_status, &zero);
+    if (status == NULL || status->state != TRACER_STATE_READY) {
+        return 0;
+    }
+
+    __u16 lport = 0;
+
+    bpf_probe_read(&lport, sizeof(lport), ((char*)newsk) + status->offset_dport + sizeof(lport));
+
+    if (lport == 0) {
+        return 0;
+    }
+
+    __u8* val = bpf_map_lookup_elem(&port_bindings, &lport);
+
+    if (val == NULL) {
+        __u8 state = PORT_LISTENING;
+
+        bpf_map_update_elem(&port_bindings, &lport, &state, BPF_ANY);
+    }
+
+    return 0;
+}
+
+SEC("kprobe/tcp_v4_destroy_sock")
+int kprobe__tcp_v4_destroy_sock(struct pt_regs* ctx) {
```

Comment:
Do we need an equivalent for v6?

Reply:
No, this is called for both v4 and v6 ([noted here](https://github.com/DataDog/datadog-process-agent/pull/240/files#diff-21d8b26d8e1cc76bc43ff4a5a6f541ddR12))